programs/perpetuals/src/instructions/set_custom_oracle_price_permissionless.rs

```diff
@@ -0,0 +1,88 @@
+//! SetCustomOraclePricePermissionless instruction handler
+
+use {
+    crate::{
+        error::PerpetualsError,
+        state::{custody::Custody, oracle::CustomOracle, perpetuals::Perpetuals, pool::Pool},
+    },
+    anchor_lang::prelude::*,
+};
+
+#[derive(Accounts)]
+pub struct SetCustomOraclePricePermissionless<'info> {
+    #[account(mut)]
+    pub user: Signer<'info>,
+
+    #[account(
+        seeds = [b"perpetuals"],
+        bump = perpetuals.perpetuals_bump
+    )]
+    pub perpetuals: Box<Account<'info, Perpetuals>>,
+
+    #[account(
+        seeds = [b"pool",
+                 pool.name.as_bytes()],
+        bump = pool.bump
+    )]
+    pub pool: Box<Account<'info, Pool>>,
+
+    #[account(
+        seeds = [b"custody",
+                 pool.key().as_ref(),
+                 custody.mint.as_ref()],
+        bump = custody.bump
+    )]
+    pub custody: Box<Account<'info, Custody>>,
+
+    #[account(
+        // Custom oracle must first be initialized by authority before permissionless updates.
+        mut,
+        seeds = [b"oracle_account",
+                 pool.key().as_ref(),
+                 custody.mint.as_ref()],
+        bump
+    )]
+    pub oracle_account: Box<Account<'info, CustomOracle>>,
+
+    system_program: Program<'info, System>,
```

Comment:
why do you need system program here?

Reply:
ah my bad, I started from a copy of `set_custom_oracle_price.rs` but don't need the system program since `init_if_needed` is no longer used.

removed! 👍

Reply:
idt you pushed this change